### PR TITLE
Wulmer fix/activate behaviour on windows

### PR DESCRIPTION
**Pull request by msarahan**
This integrates https://github.com/conda/conda-env/pull/182 along with some minor changes:

- calling activate with no arguments activates the root environment
- Conda.bat is found relative to the activate.bat script.  For all but root env, this conda.bat script will call the root conda.bat script, which should be created by conda itself in each environment as it is created.

Remaining work:
- deactivate.bat needs to use the relative conda.bat logic similarly to activate.

### CODE REVIEW

File: bin/activate.bat [hunk 1]
```diff
@@ -1,78 +1,44 @@
...
-REM First match will be the one used
+REM Use conda itself to figure things out
 
-for %%F in ("%CONDA_ENVS_PATH:;=" "%") do (
-    if exist "%%~F\%CONDA_NEW_NAME%\conda-meta" (
-       set CONDA_NEW_PATH=%%~F\%CONDA_NEW_NAME%
-       goto found_env
-    )
-)
+SET "CONDA_EXE=%~dp0\conda.bat"
```
Comment (ukoethe) [inline bin/activate.bat]: This is wrong -- the command is `conda.exe`

**Comment by ukoethe** [comment]
I tested this PR, along with https://github.com/conda/conda/pull/1727, on Windows, and it does __not__ work. I indicated some errors in the code, but there are more problems. Unfortunately, I don't have time for more detailed comments right now. Please do __not__ merge before fixing this.

**Comment by msarahan** [comment]
Thanks for testing.  I am also running into issues.  Rest assured, this will not be merged before it is working.

**Comment by msarahan** [comment]
PS: your stamp of approval would give me greater confidence in saying "it works" - I will ping you for review, if that's OK.

File: bin/activate.bat (hunk 1 above)
Comment (msarahan) [inline bin/activate.bat]: They both exist.  I think the key difference is that conda.bat might actually just contain code pointing to the root conda.bat, which then knows about conda.exe.  Conda can only exist at the root level, or else the concept of the root level is lost, and deactivating to some reference state is more ambiguous.

Comment (ukoethe) [inline bin/activate.bat]: I double-checked -- there is no `conda.bat` in my installation (or it is hiding really well :-).

Comment (msarahan) [inline bin/activate.bat]: Are you also using the development version of conda from conda/conda#1727 ? Note that you may need to manually delete/move/rename/whatever the <prefix>/Lib/site-packages/conda folder (and the conda-env folder there, too), or else the conda-installed packages take preference.  This has bitten me quite hard in debugging these packages.

Comment (ukoethe) [inline bin/activate.bat]: I did that, but still no `conda.bat`. How am I supposed to install it?

Comment (ukoethe) [inline bin/activate.bat]: Oops, now I suddenly have `conda.bat` in one of my environments - it was apparently silently installed by some other `conda install` command.

File: bin/activate.bat [hunk 2]
```diff
@@ -1,78 +1,44 @@
...
+FOR /F "delims=" %%i IN ('"%CONDA_EXE%" ..activate %CONDA_NEW_ENV%') DO set PATH=%%i
+if %ERRORLEVEL% GTR 0 (
+exit /b 1
+)
+for /F %%C IN ('"%CONDA_EXE%" ..changeps1') DO set CHANGEPS1=%%C
+if "%CHANGEPS1%" == "1" set PROMPT=[%CONDA_NEW_ENV%] $P$G
+if %ERRORLEVEL% GTR 0 (
+exit /b 1
+)
+endlocal & set PROMPT=%PROMPT%& set PATH=%PATH%& set CONDA_DEFAULT_ENV=%CONDA_NEW_ENV%
```
Comment (ukoethe) [inline bin/activate.bat]: The variable name `CONDA_DEFAULT_ENV` is now inconsistent with the code in https://github.com/conda/conda/pull/1727, where you changed it into `CONDA_ACTIVE_ENV`. As I said elsewhere, I vote for the new name.

**Comment by kalefranz** [comment]
This file has now moved to github.com/conda/conda

**Pull request closed without merging**